cli/internal/cmd/init.go

```diff
@@ -288,8 +260,9 @@ func (d *initDoer) Do(ctx context.Context) error {
 	return nil
 }
 
-func (d *initDoer) logGRPCStateChanges(ctx context.Context, wg *sync.WaitGroup, conn *grpc.ClientConn) {
+func (d *initDoer) handleGRPCStateChanges(ctx context.Context, wg *sync.WaitGroup, conn *grpc.ClientConn) {
 	go func() {
```

Comment:
Why does this function start a new Go routine instead of letting the caller handle this? @malt3

Reply:
This is almost in line with how the rest of our code handles go routines. CC @katexochen 

The general pattern we currently follow:

```go
func foo(ctx context.Context) {
  var wg sync.WaitGroup
  defer wg.Wait()
  
  usesAGoRoutine(ctx, wg)
  // do something ....
}

func usesAGoRoutine(ctx context.Context, wg sync.WaitGroup) {
  wg.Add(1)
  go func() {
    defer wg.Done()
    // do something in the background
  }
}
```